### ISSUE TITLE
Additional user-defined attributes to files.

### ISSUE THREAD

**Issue by BoPeng** [mbraak/django-file-form#314]
In our particular application we need to allow users to add additional attributes to uploaded files. On the interface side

1. A checkbox added before the filename, for users to select the "primary" file (whatever it means)
2. An input box after the filename, for users to add additional comments to files (e.g. purpose of the file).

It is not too difficult for us to fork `django-file-form` to add the fields (e.g. [this patch](https://github.com/bioworkflows/django-file-form/commit/6d4909f70513fc6f6fc3ab0f0b04bc9044ea273a) for checkbox). The most difficult part is to return attributes of new files, which need to be handled as data for the hidden field.

Other django extensions handle such cases by allowing the specification of customized models. In this case I can imaging something like

```
FILE_FORM_UPLAODED_FILE_MODEL
FILE_FORM_PLACEHOLDER_FILE_MODEL
```
to point to our models. 

The tricky part is the frontend, which needs to handle user input (checkbox and input) and add the values to hidden field, and I cannot think of a way to extend it. @mbraak do you have some idea on this? We could of course just fork `django-file-form` but would prefer contributing to the upstream whenever possible.

**Comment by mbraak** [comment]
I'll see if it's possible to add a hook in javascript to make this configurable. The checkbox patch looks like a good example for this.
The configurable model options look promising. I can have a look at that later.

**Comment by mbraak** [comment]
I'm currently working on a javascript api. One function of the api would be to manage custom attributes.

Also see #321

**Comment by BoPeng** [comment]
Implemented by #341
